src/main/scala/net/snowflake/spark/snowflake/SnowflakeConnectorUtils.scala

```diff
@@ -30,14 +30,21 @@ object SnowflakeConnectorUtils {
 
   @transient lazy val log: Logger = LoggerFactory.getLogger(getClass.getName)
 
+  def checkVersionHigherOrEqualTo(version: String, compareTo: String): Boolean = {
+    return version.split("\\.")
+      .zipAll(compareTo.split("\\."), "0", "0")
+      .find {case(a, b) => a != b }
+      .fold(0) { case (a, b) => a.toInt - b.toInt } >= 0
+  }
+
   /**
     * Check Spark version, if Spark version matches SUPPORT_SPARK_VERSION enable PushDown,
     * otherwise disable it.
     */
   val SUPPORT_SPARK_VERSION = "3.1"
 
   def checkVersionAndEnablePushdown(session: SparkSession): Boolean =
-    if (session.version.startsWith(SUPPORT_SPARK_VERSION)) {
+    if (checkVersionHigherOrEqualTo(session.version, SUPPORT_SPARK_VERSION)) {
```

Comment:
The interface classes for different spark version may be different,
So one spark connector version, we release multiple binaries for different version.
1. Don't introduce `checkVersionHigherOrEqualTo()`
2. Change `val SUPPORT_SPARK_VERSION = "3.2"`

Reply:
Got it! Done.